website/source/docs/auth/aws.html.md

```diff
@@ -0,0 +1,1006 @@
+will be the line of defense against such attacks.
+
+### Disabling reauthentication
+If a client chooses to fetch a long-lived Vault token and intends to not refresh
+(rotate) the token, then it can disable all future logins. If the option
+`disallow_reauthentication` is set, only one login will be allowed per instance.
+If the instance successfully gets the token for the first time, it can use it
+without worrying about its token getting hijacked by another entity. The client
+will still need to raise a security alert if the first login fails, since the
+backend will not be able to distinguish a genuine login attempt from an imitation,
+for the first time.
+
+When `disallow_reauthentication` option is enabled, the backend only allows a
+single successful login from the client. In this case, the client nonce loses
+its significance and hence the client can choose not to supply the nonce during
+the login.
+
+### Blacklisting role tags
+It maybe difficult to track the created role tags and to get to know which instances
+are indeed using specific role tags. In these cases, when a role tag needs to be
+blocked from any further login attempts, it can be placed in a `blacklist` via the
+endpoint `auth/aws/blacklist/roletag/<role_tag>`. Note that this will not invalidate
+the tokens that were already issued. This only blocks any further login requests.
+
+### Expiration times and tidying of `blacklist` and `whitelist` entries
+The entries in both identity `whitelist` and role tag `blacklist` are not deleted
+automatically. The entries in both of these lists will have an expiration time
+which is dynamically determined by three factors: `max_ttl` set on the AMI,
+`max_ttl` set on the role tag and `max_ttl` value of the backend mount. The
+least of these three will be set as the expiration times of these entries.
+Separate endpoints `aws/auth/whitelist/identity/tidy` and `aws/auth/blacklist/roletag/tidy`
+are provided to cleanup the entries present in these lists.
+
+### Varying public certificates
+AWS public key which is used to verify the PKCS#7 signature varies by region.
+To check if the default public certificate is applicable for the instances
+or to get a different public certificate, refer [this](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html).
+If the instances that are using this backend require more than one certificate,
+then this backend needs to be mounted at as many paths as there are certificates.
+The clients should then use appropriate mount of the backend which can verify its
+PKCS#7 signature.
+
+## Authentication
+
+### Via the CLI
+
+#### Enable AWS EC2 authentication in Vault.
+
+```
+$ vault auth-enable aws
+```
+
+#### Configure the credentials required to make AWS API calls.
+
+```
+$ vault write auth/aws/config/client secret_key=vCtSM8ZUEQ3mOFVlYPBQkf2sO6F/W7a5TVzrl3Oj access_key=VKIAJBRHKH6EVTTNXDHA region=us-east-1
```

Comment:
@issacg 

The `region` for creating the client config is extracted out of identity document and `region` is no more a parameter to `config/client` endpoint.

As Jeff mentioned earlier, the client objects once created are reused for further calls. The backend now retains the clients once it gets created for a particular region (cache is indexed by region). If the credentials are deleted or updated, the cache gets flushed.